### ISSUE TITLE
No compilation output

### ISSUE THREAD

**Issue by jderuere** [bougarfaoui/ng-socket-io#38]
(I'm using last version 0.2.0 of ng-socket-io)

Since I update to angular5 and angular-cli 1.5, I have this error when starting my app:

> ERROR in ./node_modules/ng-socket-io/socket-io.module.ts
> Module build failed: Error: /usr/src/app/node_modules/ng-socket-io/socket-io.module.ts is not part of the compilation output. Please check the other error messages for details.
>  at AngularCompilerPlugin.getCompiledFile (/usr/src/app/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:629:23)
> at plugin.done.then (/usr/src/app/node_modules/@ngtools/webpack/src/loader.js:467:39)
> at process._tickCallback (internal/process/next_tick.js:109:7)
> @ ./node_modules/ng-socket-io/index.js 3:25-54
> @ ./src/app/app.module.ts
> @ ./src/main.ts
> @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts

**Comment by Daudr** [comment]
`ng serve --aot` works

**Comment by derjp** [comment]
Make sure your tsconfig.json file looks like this:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  },
  "include": [
    "src/**/*",
    "node_modules/ng-socket-io/index.ts"
  ]
}

```
Add the index.ts file of ng-socket-io to the includes section of the file, works perfectly

**Comment by hartatovich** [comment]
@DerJP this doesn't seems to do the trick
FYI @bougarfaoui

**Comment by marnixbent** [comment]
At first it wasn't working for me, but then I noticed that I needed to restart the webserver so it compiles completely. Thanks.

**Comment by derjp** [comment]
The problem with this is that the library is incorrectly packaged. Libraries should never ship their ts files. Here's a very good explanation, I hope you can update the library accordingly as it is really helpful and should be coded properly:

https://github.com/angular/angular-cli/issues/8284#issuecomment-341417325

If this isn't updated properly even the way of including the index.ts file will break in the future. This is absolutely mandatory.

**Comment by bougarfaoui** [comment]
@Vxrize I published a new version `ng-socket-io@0.2.3` as you said, I tested it in JIT and Aot and it works just fine on both Angular and Ionic. Many thanks.

**Comment by derjp** [comment]
Great, I will test it out!
